src/main/java/picard/sam/BamErrorMetric/Pair.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package picard.sam.BamErrorMetric;
+
+/**
+ * Simple Pair class.
+ *
+ * @author Yossi Farjoun
+ */
+
+public class Pair<X extends Comparable<X>, Y extends Comparable<Y>> implements Comparable<Pair<X, Y>> {
+    private X first;
+    private Y second;
+
+    public Pair(final X x, final Y y) {
+        first = x;
+        second = y;
+    }
+
+    public X getFirst() {
+        return first;
+    }
+
+    public Y getSecond() {
+        return second;
+    }
+
+    /**
+     * Calculate whether this pair object is equal to another object.
+     *
+     * @param o The other object (hopefully a pair).
+     * @return True if the two are equal; false otherwise.
+     */
+    @Override
+    public boolean equals(final Object o) {
+        // fast return in trivial case
+        if (this == o) return true;
+
+        if (o == null) return false;
+
+        if (!(o instanceof Pair)) return false;
+
+        final Pair other = (Pair) o;
+
+        // Check to see whether one is null but not the other.
+        if (this.first == null && other.first != null) return false;
+        if (this.second == null && other.second != null) return false;
```

Comment:
this can be simplified by using `Objects.equals()`, e.g. `return Objects.equals(first, other.first) && Objects.equals(second, other.second);`